httpbakery/client.go

```diff
@@ -343,14 +344,47 @@ func (c *Client) setRequestBody(req *http.Request, body io.ReadSeeker) error {
 	if body == nil {
 		return nil
 	}
-	if req.Body == nil {
-		req.Body = ioutil.NopCloser(body)
-	} else {
-		_, err := body.Seek(0, 0)
-		if err != nil {
+	if req.Body != nil {
+		req.Body.Close()
```

Comment:
Even though that req.Body is probably going to be a `readStopper`, I'd check the error and at least log it, just in case it ever isn't.